src/App.tsx

```diff
@@ -1,54 +1,82 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  // const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isfocus, setIsfocus] = useState(false);
+
+  const filteredPeople = peopleFromServer
+    .filter(person => person.name.toLowerCase().includes(
+      appliedQuery.toLowerCase(),
+    ));
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handleClick = (name: string) => {
+    setQuery(name);
+  };
+
+  const filterdPerson: Person | undefined
+    = peopleFromServer.find(person => person.name === query);
```

Comment:
You'd better create one more state like `selectedPerson` and set person object on click instead of invoking this find method each render